### PR TITLE
Ticket2054 run diagnostics 2 page

### PR DESCRIPTION
**Pull request by Tom-Willemsen**
### Description of work

Adds a run diagnostics 2 page similar to what was present under SECI (now called "detector diagnostics")

### To test

https://github.com/ISISComputingGroup/IBEX/issues/2054

You need the changes in https://github.com/ISISComputingGroup/EPICS-isisdae/pull/19 for this to work. If you're not sure, try a `caget TE:NDW1298:DAE:DIAG:ENABLE:SP`. If it returns something then you should be good to go, if not you need to update your DAE.

The spectra data will only be supplied when the DAE is in a RUNNING state. If it looks like it's all enabled but you're not getting any data back, check that your parameters are sensible. Very large numbers in "Integral upper limit" may cause the DAE to return slowly. An example of a set of parameters for a simulated DAE that demonstrate the functionality in an appropriate way might be:
Integral lower limit: 1
Integral upper limit: 500
Maximum frames: 1000

### Acceptance criteria

- [x] Detector diagnostics page looks visually ok
- [x] Detector diagnostics page displays data from the DAE correctly. See the PVs added in https://github.com/ISISComputingGroup/EPICS-isisdae/pull/19/files to see what data should be displayed (note, you have to manually "enable" the diagnostics, this is done automatically by IBEX when the DAE perspective is opened but if you're looking at raw PVs you'll have to do it manually)
- [x] Controls are databound to the control PVs, so if the GUI is restarted the controls should have the correct numbers populated in them.

---

#### Code Review

- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/GUI-Coding-Conventions)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit tests in place? Are the unit tests small and test the a class in isolation?
- [ ] Are there automated [system tests](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/System-Testing-with-RCPTT) in place? Do they test a minimal set of functionality and leave the gui as close as possible to its original state?
- [x] Has the [manual system tests spreadsheet](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Manual-system-tests) been updated?
    - Manual system tests for the functionality should be added if there are no automated tests
    - Manual system tests can be removed from the template if they are covered by suitable automated tests
- [ ] Did any existing system test break as a result of the current changes? 
- [x] Have the changes been documented in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev). If so, do they describe the changes appropriately?
- [ ] If an OPI has been modified, does it conform to the [style guidelines](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/OPI-Creation)? There is a script called `check_opi_format.py` to help with this.

### Functional Tests

- [ ] Do changes function as described? Add comments below that describe the tests performed.
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has developer documentation been updated if required?

### Final Steps
- [x] Reviewer has moved the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev) entry for this ticket in the "Changes merged into master" section

### CODE REVIEW

File: base/uk.ac.stfc.isis.ibex.dae/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -19,12 +19,14 @@ Bundle-ActivationPolicy: lazy
...
  uk.ac.stfc.isis.ibex.dae.dataacquisition,
+ uk.ac.stfc.isis.ibex.dae.detectordiagnostics,
  uk.ac.stfc.isis.ibex.dae.experimentsetup,
  uk.ac.stfc.isis.ibex.dae.experimentsetup.periods,
  uk.ac.stfc.isis.ibex.dae.experimentsetup.timechannels,
  uk.ac.stfc.isis.ibex.dae.spectra,
  uk.ac.stfc.isis.ibex.dae.updatesettings
 Bundle-ClassPath: .
 Import-Package: org.apache.logging.log4j,
+ org.mockito.internal.util.collections,
```
Comment (mattclarke) [inline base/uk.ac.stfc.isis.ibex.dae/META-INF/MANIFEST.MF]: Does mockito need to be in here?

**Comment by mattclarke** [comment]
Suggestions for improving the UI:
* Remove 'number' from 'Starting spectrum number'
* Centering the labels over the inputs (if not too awkward to do)
* Make the tabbed area's minimum width large enough to show all inputs, and have a horizontal scroll bar for when the View is smaller than the minimum width

**Comment by mattclarke** [comment]
Switching instruments transfers the settings from the old instrument to the new one.
It should, instead, read the settings from the new one

**Comment by Tom-Willemsen** [comment]
Jenkins retest this please

**Pull request merged**